### ISSUE TITLE
apt module does not install deb from files/

### ISSUE THREAD

**Issue by admirabilis** [ansible/ansible#18901]
_From @teresaejunior on August 27, 2016 23:10_

##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

apt
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = /home/teresaejunior/system/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

```
[privilege_escalation]
become = True
```
##### OS / ENVIRONMENT

Xubuntu 16.04 with Linux 4.6-4.dmz.3-liquorix-amd64
##### SUMMARY

The convention from other core modules, like copy, unarchive and script, is to find role files under the files/ directory. The apt module does not search for file under files/, though.
##### STEPS TO REPRODUCE

```
- apt: deb=files/google-earth-stable_7.1.4.1529-r0_amd64.deb
```

##### EXPECTED RESULTS

`files/google-earth-stable_7.1.4.1529-r0_amd64.deb` should be installed
##### ACTUAL RESULTS

It says "file or directory not found".

```
TASK [earth : install google-earth-stable] *************************************
task path: /home/teresaejunior/system-test/earth/tasks/main.yml:25
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: teresaejunior
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1472339216.81-203435995230828 `" && echo ansible-tmp-1472339216.81-203435995230828="` echo $HOME/.ansible/tmp/ansible-tmp-1472339216.81-203435995230828 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpbLER8k TO /home/teresaejunior/.ansible/tmp/ansible-tmp-1472339216.81-203435995230828/apt
<127.0.0.1> EXEC /bin/sh -c 'sudo -H -S -n -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-tmqwrwbbslflzilydplhrasyhjuszxof; LANG=pt_BR.UTF-8 LC_ALL=pt_BR.UTF-8 LC_MESSAGES=pt_BR.UTF-8 /usr/bin/python /home/teresaejunior/.ansible/tmp/ansible-tmp-1472339216.81-203435995230828/apt; rm -rf "/home/teresaejunior/.ansible/tmp/ansible-tmp-1472339216.81-203435995230828/" > /dev/null 2>&1'"'"' && sleep 0'
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_args": {"allow_unauthenticated": false, "autoremove": false, "cache_valid_time": null, "deb": "files/google-earth-stable_7.1.4.1529-r0_amd64.deb", "default_release": null, "dpkg_options": "force-confdef,force-confold", "force": false, "install_recommends": null, "only_upgrade": false, "package": null, "purge": false, "state": "present", "update_cache": false, "upgrade": null}, "module_name": "apt"}, "msg": "Unable to install package: E:Não foi possível abrir arquivo files/google-earth-stable_7.1.4.1529-r0_amd64.deb - open (2: Arquivo ou diretório não encontrado)"}
```

_Copied from original issue: ansible/ansible-modules-core#4584_

**Comment by admirabilis** [comment]
_From @ansibot on August 27, 2016 23:12_



@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by admirabilis** [comment]
_From @ansibot on September 12, 2016 16:55_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by admirabilis** [comment]
_From @ansibot on October 5, 2016 15:10_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by admirabilis** [comment]
_From @ansibot on October 21, 2016 8:36_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by admirabilis** [comment]
_From @ansibot on November 6, 2016 9:35_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by admirabilis** [comment]
_From @ansibot on November 22, 2016 9:49_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by admirabilis** [comment]
_From @ansibot on December 7, 2016 18:9_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by dagwieers** [comment]
Can you please test this again with the devel branch ?
I think this is fixed either as part of a rewrite in v2.2.0 or very recently in devel.

**Comment by admirabilis** [comment]
I have just followed the [instructions](http://docs.ansible.com/ansible/intro_installation.html#running-from-source) for running the latest development version, and I still get the same error!

```
$ ansible --version
ansible 2.3.0 (devel afca957396) last updated 2016/12/14 14:39:35 (GMT -200)
  config file = /home/teresaejunior/Sistema/Projetos/xubuntu-ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

**Comment by ansibot** [comment]
cc @mgwilliams
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sgutermann** [comment]
Tested with 2.4, still a problem.

Workaround in the meantime is obviously copying the deb to the machine and executing it there.

**Comment by admirabilis** [comment]
No need to copy if you specify the full path:

    - apt:
        deb: '{{ role_path }}/files/package.deb'

**Comment by dminca** [comment]
Issue still occurring with latest Ansible by this date ( `2.4.3.0` ) ?

**Comment by mkrizek** [comment]
This is expected, from the doc [1] about the `deb` param:

_Path to a .deb package on the **remote machine**._

I suspect your `files` directory is on the controller? Your are supposed to copy the file on the remote machine in a separate task.

[1] http://docs.ansible.com/ansible/latest/modules/apt_module.html

**Comment by admirabilis** [comment]
Indeed this is in a project for local provisioning! I hadn't paid attention to this back in the day!
